core/src/test/java/org/apache/iceberg/TestRemoveSnapshots.java

```diff
@@ -424,7 +432,11 @@ public void testRetainNAvailableSnapshotsWithTransaction() {
 
     // Retain last 2 snapshots
     Transaction tx = table.newTransaction();
-    tx.expireSnapshots().expireOlderThan(t3).retainLast(2).commit();
+    removeSnapshots(tx.table())
+        .withIncrementalCleanup(incrementalCleanup)
```

Comment:
Why call `withIncrementalCleanup` here? Isn't this already called by `removeSnapshots`?

Reply:
Updated